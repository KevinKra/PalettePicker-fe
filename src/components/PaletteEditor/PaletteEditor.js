import React, { Fragment } from "react";
import "./PaletteEditor.css";

export default function PaletteEditor() {
  return (
    <Fragment>
      <div className="screen" />
      <section className="PaletteEditor">
        <button className="editor-exit">X</button>
        <form className="editor-form" action="">
          <h3 className="editor-title">Title</h3>
          <label htmlFor="">Label 1</label>
          <input type="text" />
          <label htmlFor="">Label 2</label>
          <input type="text" />
        </form>
      </section>
    </Fragment>
  );
}
