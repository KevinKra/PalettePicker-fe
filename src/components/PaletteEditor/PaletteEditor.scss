.PaletteEditor {
  position: fixed;
  margin-left: auto;
  margin-right: auto;
  top: 30%;
  left: 15%;
  right: 15%;
  z-index: 4;
  max-width: 500px;
  height: 40vh;
  border: 1px solid red;
  color: red;
}

.editor-form {
  background-color: white;
  height: 100%;
}

.editor-title {
  margin: 0;
}

.editor-exit {
  position: absolute;
  right: -10px;
  top: -10px;
}

.screen {
  position: absolute;
  z-index: 3;
  background-color: rgba(6, 16, 20, 0.658);
  height: 100%;
  width: 100vw;
}
