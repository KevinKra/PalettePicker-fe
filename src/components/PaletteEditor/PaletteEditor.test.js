import React from "react";
import PaletteEditor from "./PaletteEditor";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("PaletteEditor", () => {
  let wrapper;
  const historyMock = { push: jest.fn() };
  beforeEach(() => {
    wrapper = shallow(<PaletteEditor history={historyMock} />);
  });
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should return to the previous page when 'editor-exit' has been clicked", () => {
    const button = wrapper.find(".editor-exit");
    button.simulate("click");
    expect(historyMock.push.mock.calls[0]).toEqual(["/"]);
  });
});
