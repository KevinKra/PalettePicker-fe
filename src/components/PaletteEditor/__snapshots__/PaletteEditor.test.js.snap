// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteEditor should match the snapshot 1`] = `
<Fragment>
  <div
    className="screen"
  />
  <section
    className="PaletteEditor"
  >
    <button
      className="editor-exit"
      onClick={[Function]}
    >
      X
    </button>
    <form
      action=""
      className="editor-form"
    >
      <h3
        className="editor-title"
      >
        Title
      </h3>
      <label
        htmlFor=""
      >
        Label 1
      </label>
      <input
        type="text"
      />
      <label
        htmlFor=""
      >
        Label 2
      </label>
      <input
        type="text"
      />
    </form>
  </section>
</Fragment>
`;
